### PR TITLE
Implement service creation 

### PR DESCRIPTION
**Pull request by folex**

### CODE REVIEW

File: crates/ivalue-utils/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+)]
+
+pub use wasmer_wit::types::InterfaceType as IType;
+pub use wasmer_wit::values::InterfaceValue as IValue;
+pub use wasmer_wit::vec1;
+
+use serde_json::{json, Value};
+use vec1::Vec1;
+
+pub fn into_str(v: &IValue) -> Option<&str> {
```
Comment (michaelvoronov) [inline crates/ivalue-utils/src/lib.rs:35]: `as_str` is more convenient with Rust style guide. Prefix `into` says that value should be moved.

File: crates/ivalue-utils/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+pub fn into_string(v: IValue) -> Option<String> {
+    if let IValue::String(s) = v {
+        Some(s)
+    } else {
+        None
+    }
+}
+
+/// Converts result of call_service into `IValue::Record`
+pub fn as_record_opt(v: std::result::Result<Option<Value>, Value>) -> Option<IValue> {
```
Comment (michaelvoronov) [inline crates/ivalue-utils/src/lib.rs:52]: `into_record_opt`

File: crates/ivalue-utils/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    match v {
+        Ok(None) => unit(),
+        Ok(Some(v)) => ok(v),
+        Err(e) => error(e),
+    }
+}
+
+/// Converts result of call_service into `IValue::Record`
+pub fn as_record(v: std::result::Result<Value, Value>) -> Option<IValue> {
```
Comment (michaelvoronov) [inline crates/ivalue-utils/src/lib.rs:61]: `into_record`

File: crates/json-utils/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    unused_mut,
+    unused_variables,
+    unused_unsafe,
+    unreachable_patterns
+)]
+
+use serde_json::Value;
+use std::fmt::Debug;
+
+pub fn into_string(v: Value) -> Option<String> {
```
Comment (michaelvoronov) [inline crates/json-utils/src/lib.rs:31]: Suggest to call this Value as `JValue` everywhere for easy understanding of code.

**Review by michaelvoronov — APPROVED** [review]

File: crates/ivalue-utils/src/lib.rs (hunk 2 above)
Comment (folex) [reply crates/ivalue-utils/src/lib.rs:52]: ok, I'll rename it in the next PR if you don't mind

File: crates/ivalue-utils/src/lib.rs (hunk 3 above)
Comment (folex) [reply crates/ivalue-utils/src/lib.rs:61]: ok, I'll rename it in the next PR if you don't mind

File: crates/json-utils/src/lib.rs (hunk 4 above)
Comment (folex) [reply crates/json-utils/src/lib.rs:31]: ok, I'll rename it in the next PR if you don't mind

Comment (folex) [reply crates/json-utils/src/lib.rs:31]: ok, I'll rename it in the next PR if you don't mind

**Pull request merged**